Wire U6, the nodal B-bar volumetric patch element, into ccx.

Adds the dispatcher branches, mastruct registration and build entries for
elements_ccx/e_c3d_u6.f, u6patch.f and resultsmech_u6.f. Also threads lakon
and ne through e_c3d_u, which U6 needs to find the tets around its centre
node -- the patch geometry cannot be recovered from a node list alone.

--- e_c3d_u.f	2026-08-22 11:20:26.682935314 +0300
+++ e_c3d_u.f	2026-08-22 17:43:18.176803547 +0300
@@ -17,7 +17,8 @@
 !     Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 !     
       subroutine e_c3d_u(co,kon,lakonl,p1,p2,omx,bodyfx,nbody,s,sm,
-     &     ff,nelem,nmethod,elcon,nelcon,rhcon,nrhcon,alcon,nalcon,
+     &     ff,nelem,lakon,ne,nmethod,elcon,nelcon,rhcon,nrhcon,alcon,
+     &     nalcon,
      &     alzero,ielmat,ielorien,norien,orab,ntmat_,
      &     t0,t1,ithermal,vold,iperturb,nelemload,
      &     sideload,xload,nload,idist,sti,stx,iexpl,plicon,
@@ -47,11 +48,12 @@
 !     
       integer mass(*),stiffness,buckling,rhsi,coriolis
 !     
-      character*8 lakonl
+      character*8 lakonl,lakon(*)
       character*20 sideload(*)
       character*80 matname(*),amat
       character*81 tieset(3,*)
 !     
+      integer ne
       integer konl(20),nelemload(2,*),nbody,nelem,mi(*),kon(*),
      &     ielprop(*),null,index,mattyp,ithermal(*),iperturb(*),nload,
      &     idist,
@@ -88,6 +90,45 @@
      &       ne0,ipkon,thicke,integerglob,doubleglob,tieset,istartset,
      &       iendset,ialset,ntie,nasym,ielprop,prop)
 !        
+      elseif(lakonl(2:2).eq.'6') then
+!
+!       SPAX: U6, nodal-averaged B-bar volumetric patch
+!
+        call e_c3d_u6(co,kon,lakonl,s,ff,nelem,elcon,nelcon,
+     &       ielmat,mi,ncmat_,ntmat_,ipkon,lakon,ne,stiffness)
+!
+      elseif(lakonl(2:2).eq.'5') then
+!
+!       SPAX: U5, deviatoric-only tet (nodal-averaged B-bar)
+!
+        call e_c3d_u5(co,kon,lakonl,p1,p2,omx,bodyfx,nbody,s,sm,
+     &       ff,nelem,nmethod,elcon,nelcon,rhcon,nrhcon,alcon,nalcon,
+     &       alzero,ielmat,ielorien,norien,orab,ntmat_,
+     &       t0,t1,ithermal,vold,iperturb,nelemload,
+     &       sideload,xload,nload,idist,sti,stx,iexpl,plicon,
+     &       nplicon,plkcon,nplkcon,xstiff,npmat_,dtime,
+     &       matname,mi,ncmat_,mass,stiffness,buckling,rhsi,intscheme,
+     &       ttime,time,istep,iinc,coriolis,xloadold,reltime,
+     &       ipompc,nodempc,coefmpc,nmpc,ikmpc,ilmpc,veold,
+     &       ne0,ipkon,thicke,integerglob,doubleglob,tieset,istartset,
+     &       iendset,ialset,ntie,nasym,ielprop,prop)
+!
+      elseif(lakonl(2:2).eq.'4') then
+!
+!       SPAX: U4, mixed displacement/pressure tetrahedron (MINI, P1+bubble/P1)
+!
+        call e_c3d_u4(co,kon,lakonl,p1,p2,omx,bodyfx,nbody,s,sm,
+     &       ff,nelem,nmethod,elcon,nelcon,rhcon,nrhcon,alcon,nalcon,
+     &       alzero,ielmat,ielorien,norien,orab,ntmat_,
+     &       t0,t1,ithermal,vold,iperturb,nelemload,
+     &       sideload,xload,nload,idist,sti,stx,iexpl,plicon,
+     &       nplicon,plkcon,nplkcon,xstiff,npmat_,dtime,
+     &       matname,mi,ncmat_,mass,stiffness,buckling,rhsi,intscheme,
+     &       ttime,time,istep,iinc,coriolis,xloadold,reltime,
+     &       ipompc,nodempc,coefmpc,nmpc,ikmpc,ilmpc,veold,
+     &       ne0,ipkon,thicke,integerglob,doubleglob,tieset,istartset,
+     &       iendset,ialset,ntie,nasym,ielprop,prop)
+!
       elseif(lakonl(2:4).eq.'S45') then
         call e_c3d_us45(co,kon,lakonl,p1,p2,omx,bodyfx,nbody,s,sm,
      &       ff,nelem,nmethod,elcon,nelcon,rhcon,nrhcon,alcon,nalcon,
--- resultsmech_u.f	2026-08-22 11:20:26.684935397 +0300
+++ resultsmech_u.f	2026-08-22 17:43:18.183801148 +0300
@@ -68,6 +68,30 @@
      &        ncmat_,nstate_,stiini,vini,ener,eei,enerini,istep,iinc,
      &        reltime,calcul_fn,calcul_qa,calcul_cauchy,nener,
      &        ikin,nal,ne0,thicke,emeini,nelem,ielprop,prop,t0g,t1g)
+      elseif(lakon(nelem)(2:2).eq.'6') then
+!
+!     SPAX: U6 volumetric patch -- internal forces only, no stress
+!
+         call resultsmech_u6(co,kon,ipkon,lakon,ne,v,elcon,nelcon,
+     &        ielmat,mi,ncmat_,ntmat_,fn,calcul_fn,nal,nelem)
+      elseif(lakon(nelem)(2:2).eq.'5') then
+!
+!     SPAX: U5 deviatoric tet
+!
+         call resultsmech_u5(co,kon,ipkon,lakon,v,stx,elcon,nelcon,
+     &        rhcon,nrhcon,alcon,nalcon,alzero,ielmat,ielorien,norien,
+     &        orab,ntmat_,t0,t1,ithermal,iperturb,fn,iout,qa,vold,
+     &        nmethod,dtime,plicon,nplicon,plkcon,nplkcon,xstiff,
+     &        npmat_,matname,mi,ncmat_,calcul_fn,calcul_qa,nal,nelem)
+      elseif(lakon(nelem)(2:2).eq.'4') then
+!
+!     SPAX: U4 mixed displacement/pressure tetrahedron
+!
+         call resultsmech_u4(co,kon,ipkon,lakon,v,stx,elcon,nelcon,
+     &        rhcon,nrhcon,alcon,nalcon,alzero,ielmat,ielorien,norien,
+     &        orab,ntmat_,t0,t1,ithermal,iperturb,fn,iout,qa,vold,
+     &        nmethod,dtime,plicon,nplicon,plkcon,nplkcon,xstiff,
+     &        npmat_,matname,mi,ncmat_,calcul_fn,calcul_qa,nal,nelem)
       elseif(lakon(nelem)(2:4).eq.'S45') then
 !
 !     user element us45 flat shell element with 4 nodes (5dof)
--- mastruct.c	2026-08-22 11:24:54.127557241 +0300
+++ mastruct.c	2026-08-22 17:43:18.184999192 +0300
@@ -127,6 +127,9 @@
 	indexe=ipkon[i];
 	if(strcmp1(&lakon[8*i],"U")==0){
 	  if((strcmp1(&lakon[8*i+1],"1")==0)||
+	     (strcmp1(&lakon[8*i+1],"4")==0)||
+	     (strcmp1(&lakon[8*i+1],"5")==0)||
+	     (strcmp1(&lakon[8*i+1],"6")==0)||
 	     (strcmp1(&lakon[8*i+1],"S45")==0)||
 	     (strcmp1(&lakon[8*i+1],"S3")==0)){
 	  
@@ -358,7 +361,10 @@
       indexe=ipkon[i];
       if(strcmp1(&lakon[8*i],"U")==0){
 	if((strcmp1(&lakon[8*i+1],"1")==0)||
-	   (strcmp1(&lakon[8*i+1],"S45")==0)||
+	   (strcmp1(&lakon[8*i+1],"4")==0)||
+	     (strcmp1(&lakon[8*i+1],"5")==0)||
+	     (strcmp1(&lakon[8*i+1],"6")==0)||
+	     (strcmp1(&lakon[8*i+1],"S45")==0)||
 	   (strcmp1(&lakon[8*i+1],"S3")==0)){
 	  
 	  /* user element
--- mastructcs.c	2026-08-22 11:24:54.128557286 +0300
+++ mastructcs.c	2026-08-22 17:43:18.186067522 +0300
@@ -69,7 +69,10 @@
     indexe=ipkon[i];
     if(strcmp1(&lakon[8*i],"U")==0){
       if((strcmp1(&lakon[8*i+1],"1")==0)||
-	 (strcmp1(&lakon[8*i+1],"S45")==0)||
+	 (strcmp1(&lakon[8*i+1],"4")==0)||
+	     (strcmp1(&lakon[8*i+1],"5")==0)||
+	     (strcmp1(&lakon[8*i+1],"6")==0)||
+	     (strcmp1(&lakon[8*i+1],"S45")==0)||
 	 (strcmp1(&lakon[8*i+1],"S3")==0)){
 	  
 	/* user element
@@ -184,7 +187,10 @@
     indexe=ipkon[i];
     if(strcmp1(&lakon[8*i],"U")==0){
       if((strcmp1(&lakon[8*i+1],"1")==0)||
-	 (strcmp1(&lakon[8*i+1],"S45")==0)||
+	 (strcmp1(&lakon[8*i+1],"4")==0)||
+	     (strcmp1(&lakon[8*i+1],"5")==0)||
+	     (strcmp1(&lakon[8*i+1],"6")==0)||
+	     (strcmp1(&lakon[8*i+1],"S45")==0)||
 	 (strcmp1(&lakon[8*i+1],"S3")==0)){
 	  
 	/* user element
--- Makefile.inc	2026-08-22 11:20:50.209913322 +0300
+++ Makefile.inc	2026-08-22 17:43:18.186826118 +0300
@@ -275,6 +275,14 @@
 e_c3d_se.f \
 e_c3d_th.f \
 e_c3d_u1.f \
+	e_c3d_u4.f \
+	e_c3d_u5.f \
+	e_c3d_u6.f \
+	u6patch.f \
+	resultsmech_u6.f \
+	resultsmech_u5.f \
+	u4mat.f \
+	resultsmech_u4.f \
 e_c3d_us3.f \
 e_c3d_us45.f \
 e_c3d_u.f \
